homeassistant/components/light/limitlessled.py

```diff
@@ -159,30 +159,44 @@ class LimitlessLEDGroup(Light):
 
     def __init__(self, group, config):
         """Initialize a group."""
-        self.group = group
-        self.repeating = False
-        self._is_on = False
-        self._brightness = None
-        self.config = config
-
-    @staticmethod
-    def factory(group, config):
-        """Produce LimitlessLEDGroup objects."""
         from limitlessled.group.rgbw import RgbwGroup
         from limitlessled.group.white import WhiteGroup
         from limitlessled.group.rgbww import RgbwwGroup
         if isinstance(group, WhiteGroup):
-            return LimitlessLEDWhiteGroup(group, config)
+            self._supported = SUPPORT_LIMITLESSLED_WHITE
         elif isinstance(group, RgbwGroup):
-            return LimitlessLEDRGBWGroup(group, config)
+            self._supported = SUPPORT_LIMITLESSLED_RGB
         elif isinstance(group, RgbwwGroup):
-            return LimitlessLEDRGBWWGroup(group, config)
+            self._supported = SUPPORT_LIMITLESSLED_RGBWW
+
+        self.group = group
+        self.config = config
+        self.repeating = False
+        self._is_on = False
+        self._brightness = None
+        self._temperature = None
+        self._color = None
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Called when entity is about to be added to hass."""
+        last_state = yield from async_get_last_state(self.hass, self.entity_id)
```

Comment:
Note: since LimitlessLED doesn't implement a unique ID, we can't guarantee that the entity ID is unique after each restart.

Reply:
Noted. Actually I hadn't thought of that but it is just another case where the assumed state can get wrong, nothing fatal.